packages/core/core/src/utils.ts

```
@@ -94,3 +100,80 @@ export function useSlotProps<T extends string>(
+    if (variant in props) {
+      // Only add the key if it's actually in T
+      (acc as any)[variant as keyof T] = props[variant as keyof T];
+    }
+    return acc;
+  }, {} as Partial<T>);
+
+  if (filter) {
+    const omitted = Object.keys(props)
+      .filter(key => !variants.includes(key as K))
+      // eslint-disable-next-line unicorn/no-array-reduce
+      .reduce((acc, key) => ({ ...acc, [key]: props[key] }), {} as Partial<T>);
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)